code/network/multi_pxo.cpp

```diff
@@ -2320,35 +2271,17 @@ void multi_pxo_get_channels()
  */
 void multi_pxo_clear_channels()
 {
-	pxo_channel *moveup,*backup;
-	
 	// only clear a non-null list
-	if(Multi_pxo_channels != NULL){		
-		// otherwise
-		moveup = Multi_pxo_channels;
-		backup = NULL;
-		if(moveup != NULL){
-			do {			
-				backup = moveup;
-				moveup = moveup->next;			
-		
-				// free the struct itself
-				vm_free(backup);
-				backup = NULL;
-			} while(moveup != Multi_pxo_channels);
-			Multi_pxo_channels = NULL;
-		}	
-
-		// head of the list of available channels
-		Multi_pxo_channels = NULL;
-		Multi_pxo_channel_count = 0;
+	if(Multi_pxo_channels.empty()){		
```

Comment:
That check needs to be negated, since you want to do that stuff if it's *not* empty.